data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/export/ExportScheduler.java

```diff
@@ -110,24 +129,52 @@ private boolean createDataQueryPartitions(final String collection,
 
         if (totalQueries.get() > 0) {
             exportPartitionTotalCounter.increment(totalQueries.get());
+            final ExportLoadStatus exportLoadStatus = ExportLoadStatus.fromMap(globalState.getProgressState().get());
+            totalQueries.getAndAdd(exportLoadStatus.getTotalPartitions());
+            exportLoadStatus.setTotalPartitions(totalQueries.get());
+            exportLoadStatus.setLastUpdateTimestamp(Instant.now().toEpochMilli());
+            globalState.setProgressState(exportLoadStatus.toMap());
 
             // Currently, we need to maintain a global state to track the overall progress.
             // So that we can easily tell if all the export files are loaded
-            final ExportLoadStatus exportLoadStatus = new ExportLoadStatus(totalQueries.get(), 0, 0, Instant.now().toEpochMilli());
-            enhancedSourceCoordinator.createPartition(new GlobalState(EXPORT_PREFIX + collection, exportLoadStatus.toMap()));
+            enhancedSourceCoordinator.saveProgressStateForPartition(globalState, null);
             return true;
         } else {
             return false;
         }
     }
 
+    private void updateExportPartition(final ExportPartition exportPartition,
+                                       final PartitionIdentifierBatch partitionIdentifierBatch) {
+        final ExportProgressState state = exportPartition.getProgressState().get();
+        if (partitionIdentifierBatch.getEndDocId() != null) {
+            state.setLastEndDocId(partitionIdentifierBatch.getEndDocId());
+            enhancedSourceCoordinator.saveProgressStateForPartition(exportPartition, null);
+        }
+    }
+
     private void completeExportPartition(final ExportPartition exportPartition) {
         exportJobSuccessCounter.increment();
         final ExportProgressState state = exportPartition.getProgressState().get();
         state.setStatus(COMPLETED_STATUS);
         enhancedSourceCoordinator.completePartition(exportPartition);
     }
 
+    private void markTotalPartitionsAsComplete(final String collection) {
+        final String exportPartitionKey = EXPORT_PREFIX + collection;
+        Optional<EnhancedSourcePartition> globalPartition = enhancedSourceCoordinator
+                .getPartition(exportPartitionKey);
+        if (globalPartition.isEmpty()) {
```

Comment:
Is this check redundant? There is a wait on global partition already around Line 78.

Reply:
I agree at this point the check is redundant